db/migrate/20160321160404_add_middleware_event_streams.rb

```diff
@@ -0,0 +1,7 @@
+class AddMiddlewareEventStreams < ActiveRecord::Migration[5.0]
+  def change
+    add_column :event_streams, :middleware_type, :string   # relevant type (entity class name)
+    add_column :event_streams, :middleware_id, :bigint     # MIQ entity id
```

Comment:
@Fryguy @abonas Right, that is what I would have used if I was trying to add columns to the db for every inventory type that could be relevant to a hawkular event.  I started by looking at Alissa's container stuff where I think they provided fields for 3 hardcoded types, so I saw what had been done.  I can still do this, but I was trying to get away from that approach and try something more generic.  As the middleware provider grows it's conceivable we could have quite a lot of types, many more than 3.  RHQ had hundreds.  if you prefer that I stick with the hardcoded approach I'll go back in that direction, perhaps there is something downstream that I am missing, but using middleware_id, middleware_name and inferring the relevant type via middleware_type could save a lot of PRs, table growth, and db migrations.  Let me know. I'm free to talk it over in a call as well.